packages/nhost-js/src/utils/helpers.ts

```diff
@@ -1,7 +1,8 @@
 import { NhostClientConstructorParams } from './types'
 
 // a port can be a number or a placeholder string with leading and trailing double underscores, f.e. "8080" or "__PLACEHOLDER_NAME__"
-const LOCALHOST_REGEX = /^((?<protocol>http[s]?):\/\/)?(?<host>localhost)(:(?<port>(\d+|__\w+__)))?$/
+const LOCALHOST_REGEX =
```

Comment:
TBH, I'm not sure if my implementation makes sense as is.

Some context for you @plmercereau:

Now that `https://localdev.nhost.run` will be used by the local dashboard, I had to change these things here to prevent the error caused by the region check on line 51.

I didn't want to include `.nhost.run` in the subdomain configuration of the dashboard, because that's not how we would initialize an `NhostClient` in a project that's not our dashboard. I thought it makes more sense to just specify it as `localdev`, but then the RegExp looks a bit weird to me. I'd appreciate your thoughts on this.

Reply:
> I didn't want to include `.nhost.run` in the subdomain configuration of the dashboard, because that's not how we would initialize an `NhostClient` in a project that's not our dashboard. I thought it makes more sense to just specify it as `localdev`

I don't understand this part

Reply:
For `localhost`, we don't need the `.nhost.run` ending, but we do need it in every other case. Subdomain is passed to `NhostClient` as a simple subdomain without the `.nhost.run`. I just didn't want to make an exception here to pass it like `subdomain: 'localdev.nhost.run'` so I need additional checks inside the SDK and I need to add it manually for localdev.

Reply:
Thanks, I got it.
One last thing that I don't get: why don't we need a port when using `localdev.nhost.run`? How traefik listen to ports 80/443?

Reply:
The former port `1337` becomes `443` and yet it will be configurable (in order to run multiple nhost apps)